### PR TITLE
[Draft] 86 metadata definition UI

### PR DESCRIPTION
**Pull request by arunapa**

### CODE REVIEW

**Comment by arunapa** [comment]
Screen Shot 2022-09-30 at 11 07 21 AM
Screen Shot 2022-09-30 at 11 07 37 AM

File: frontend/src/components/navigation/TopBar.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ export default function TopBar() {
 					<Box display="flex" flexGrow={1}>
 						<img src="../../public/clowder-logo-sm.svg" alt="clowder-logo-sm"/>
 						<Link href="/" sx={link}>Explore</Link>
+						<Link href="/new-metadata-definition" sx={link}>Create New Metadata</Link>
```
Comment (longshuicy) [inline frontend/src/components/navigation/TopBar.tsx:36]: This seems too wordy. Maybe just say New Metadata

File: frontend/src/components/metadata/CreateMetadataDefinition.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+							fullWidth
+							id="metadataDescription"
+							label="Metadata Description"
+							name="metadataDescription"
+							value={metadataDescription}
+							onChange={(event) => { setMetadataDescription(event.target.value); }}
+						/>
+					</Grid>
+					<Grid xs={12} md={8}>
+						<Button onClick={() => {postMetadata();}}>+ Add New Field</Button>
```
Comment (longshuicy) [inline frontend/src/components/metadata/CreateMetadataDefinition.tsx:96]: I'm thinking at least you show one group of input box; then add a plus button to duplicate the same group.
This group should at least include:
- name input box
- widgetType dropdown
- type dropdown
- required (toggle button or radio button)

File: frontend/src/components/metadata/CreateMetadataDefinition.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		setMetadataName("");
+		setMetadataDescription("");
+	}
+
+	return (
+		<div className="outer-container">
+			<div className="inner-container">
+				<Grid container spacing={2} rowSpacing={4}>
+					<Grid xs={12} md={8}>
+						<TextField
```
Comment (longshuicy) [inline frontend/src/components/metadata/CreateMetadataDefinition.tsx:68]: You can use styled clowder text field this way it will take the full width
See `frontend/src/components/styledComponents/ClowderInput.jsx` or `frontend/src/components/styledComponents/ClowderMetadataTextField.jsx`

Comment (longshuicy) [reply frontend/src/components/metadata/CreateMetadataDefinition.tsx:68]: image
 
 right now the input/form didn't take the full width of the dialog

Comment (longshuicy) [reply frontend/src/components/metadata/CreateMetadataDefinition.tsx:68]: Sorry for nitpicking... the cancel on the bottom right seems a little bit odd (?). Maybe "x" on the top right might make more sense?

**Review by longshuicy — CHANGES_REQUESTED** [review]

**Comment by longshuicy** [comment]
Can I close this? This is duplicate from https://github.com/clowder-framework/clowder2/pull/150
In the future you can click this button to flip the status instead of creating new PR
image
